bouncer.go

```diff
@@ -266,6 +315,157 @@ type Login struct {
 	Expire string `json:"expire"`
 }
 
+type SiteVerifyResponse struct {
+	Success     bool      `json:"success"`
+	Action      string    `json:"action"`
+	ChallengeTS time.Time `json:"challenge_ts"`
+	Hostname    string    `json:"hostname"`
+	ErrorCodes  []string  `json:"error-codes"`
+}
+
+type ValidationResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+}
+
+func handleBanResponseSoft(bouncer *Bouncer, rw http.ResponseWriter, req *http.Request) {
+	if bouncer.forbidOnFailure {
+		rw.WriteHeader(http.StatusForbidden)
+		return
+	}
+	bouncer.next.ServeHTTP(rw, req)
+}
+
+func handleBanResponseForced(bouncer *Bouncer, rw http.ResponseWriter, req *http.Request) {
+	rw.WriteHeader(http.StatusForbidden)
+	return
+}
+
+func handleBanResponseOrCaptcha(bouncer *Bouncer, rw http.ResponseWriter, req *http.Request, isCaptcha bool) {
+	logger.Debug(fmt.Sprintf("handleBanResponseOrCaptcha '%t'", isCaptcha))
+	handleBanResponseForced(bouncer, rw, req)
+
+	if !isCaptcha {
+		return
+	}
+
+	logger.Debug(fmt.Sprintf("Must respond with captcha here.... Captcha file '%s'", bouncer.captchaHtmlFilePath))
+	content, err := os.ReadFile(bouncer.captchaHtmlFilePath)
```

Comment:
you can declare the captchaPage in the global context, not need to read the file every time you send it